tools/rsyslogd.c

```diff
@@ -41,6 +41,9 @@
 #ifdef ENABLE_LIBCAPNG
 	#include <cap-ng.h>
 #endif
+#if defined(HAVE_CLOSE_RANGE)
+#	include <linux/close_range.h>
+#endif
```

Comment:
Just to confirm, this is decided at build time right? Will that cause any issues on systems when it's not available at runtime?

I'm not familiar with C, just curious if the logic below fails gracefully if that's the case. Containers can be built with software on one system and run on many others where the kernel is different.

Reply:
Well, the include is build-time only. The systemcall obviously is runtime. If the loader can't load rsyslog due to missing kernel/libc symbols, that's something the packagers need to decide. If the symbol loads and the call fails, we again go to the loop.